features/unit/v2indexerclient_paths.feature

```diff
@@ -183,6 +183,29 @@ Feature: Indexer Client v2 Paths
       | path                  | application-id |
       | /v2/applications/1234 | 1234           |
 
+  @unit.applications.boxes
+  Scenario Outline: LookupApplicationBoxByIDandName path
+    When we make a LookupApplicationBoxByIDandName call with applicationID <application-id> with encoded box name "<encoded-box-name>"
+    Then expect the path used to be "<path>"
+
+    Examples:
+      | path                                                | application-id | encoded-box-name     |
+      | /v2/applications/1234/box?name=b64%3AaGVsbG8%3D     | 1234           | b64:aGVsbG8=         |
+      | /v2/applications/1234/box?name=b64%3A%2Fw%3D%3D     | 1234           | b64:/w==             |
+      | /v2/applications/1234/box?name=b64%3A8J%2BSqQ%3D%3D | 1234           | b64:8J+SqQ==         |
+      | /v2/applications/1234/box?name=b64%3AYS96           | 1234           | b64:YS96             |
+
+  @unit.applications.boxes
+  Scenario Outline: SearchForApplicationBoxes path
+    When we make a SearchForApplicationBoxes call with applicationID <application-id> with max <max> nextToken "<nextToken>"
+    Then expect the path used to be "<path>"
+
+    Examples:
+      | path                                           | application-id | max | nextToken |
+      | /v2/applications/1234/boxes                    | 1234           | 0   |           |
+      | /v2/applications/1234/boxes?limit=2            | 1234           | 2   |           |
+      | /v2/applications/1234/boxes?limit=2&next=token | 1234           | 2   | token     |
```

Comment:
If we assume that the token is passed directly back (which is my opinion) then this works well. If we assume that the the SDK tries to parse `nextToken` and then reproduce it `b64` encoded, then more complex unit tests would be called for. Cases such as:
1. `nextToken=b64:bmV4dEJveA==` produces `/v2/applications/1234/boxes?limit=2&next=b64:bmV4dEJveA==`
2. `nextToken=str:nextBox` produces `/v2/applications/1234/boxes?limit=2&next=b64:bmV4dEJveA==`
But again, I don't favor this more complex approach.

Reply:
The previous comment is misleading. Let me try again. Let's look at [this autogenerated fixture](https://github.com/algorand/indexer/blob/98df20f7a0fb1c515f2faa25de7579aa429234d3/api/test_resources/boxes.json#L764).

In particular, the request url is:
```url
http://localhost:8999/v2/applications/1/boxes?limit=3\u0026next=b64%3AYm94ICM4
```
and the response was

```json
{
    "statusCode": 200,
    "body": "{\"application-id\":1,\"boxes\":[{\"name\":\"ZGlzYXBwb2ludGluZyBib3g=\"},{\"name\":\"ZG9uJ3QgYm94IG1lIGluIHRoaXMgd2F5\"},{\"name\":\"ZmFudGFidWxvdXM=\"}],\"next-token\":\"ZmFudGFidWxvdXM=\"}\n"
},
```
So really what this example is stating is that if the _most recently received_ token was `token`, we should therefore supply `&next=b64:token` (and that is strange because `token` is not a valid b64 encoding)

Reply:
The previous comment is misleading. Let me try again. Let's look at [this autogenerated fixture](https://github.com/algorand/indexer/blob/98df20f7a0fb1c515f2faa25de7579aa429234d3/api/test_resources/boxes.json#L764).

In particular, the request url is:
```url
http://localhost:8999/v2/applications/1/boxes?limit=3\u0026next=b64%3AYm94ICM4
```
and the response was

```json
{
    "statusCode": 200,
    "body": "{\"application-id\":1,\"boxes\":[{\"name\":\"ZGlzYXBwb2ludGluZyBib3g=\"},{\"name\":\"ZG9uJ3QgYm94IG1lIGluIHRoaXMgd2F5\"},{\"name\":\"ZmFudGFidWxvdXM=\"}],\"next-token\":\"ZmFudGFidWxvdXM=\"}\n"
},
```
So really what this example is stating is that if the _most recently received_ token was `token`, we should therefore supply `&next=b64:token` (and that is strange becose `token` is not a valid b64 encoding)


More context in [the swagger](https://github.com/algorand/indexer/blob/3457a5ace6d56f2b76584335547c73729e889f96/api/indexer.oas2.json#L3048)

Reply:
Hey, from our previous discussion, I remembered we concluded that we would accept b64 encoded `next` token, and I think the testcases on `next` token should be modified?

Reply:
am resolving and gonna put a new comment entirely in place of all these stale statements